providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/grpc/RpcResolver.java

```diff
@@ -61,40 +73,53 @@ public final class GrpcResolver implements Resolver {
      * @param cache             cache to use
      * @param onProviderEvent lambda which handles changes in the connection/stream
      */
-    public GrpcResolver(
+    public RpcResolver(
             final FlagdOptions options, final Cache cache, final Consumer<FlagdProviderEvent> onProviderEvent) {
         this.cache = cache;
         this.strategy = ResolveFactory.getStrategy(options);
         this.options = options;
-        this.connector = new GrpcConnector<>(
-                options,
-                ServiceGrpc::newStub,
-                ServiceGrpc::newBlockingStub,
-                onProviderEvent,
-                stub -> stub.eventStream(
-                        Evaluation.EventStreamRequest.getDefaultInstance(),
-                        new EventStreamObserver(
-                                (flags) -> {
-                                    if (this.cache != null) {
-                                        flags.forEach(this.cache::remove);
-                                    }
-                                    onProviderEvent.accept(new FlagdProviderEvent(
-                                            ProviderEvent.PROVIDER_CONFIGURATION_CHANGED, flags));
-                                },
-                                onProviderEvent)));
+        streamReceiver = new LinkedBlockingQueue<>(QUEUE_SIZE);
+        this.connector = new ChannelConnector<>(options, ServiceGrpc::newBlockingStub, onProviderEvent);
+        this.onProviderEvent = onProviderEvent;
     }
 
     /**
      * Initialize Grpc resolver.
      */
     public void init() throws Exception {
         this.connector.initialize();
+
+        Thread listener = new Thread(() -> {
+            try {
+                observeEventStream(shutdown);
```

Comment:
This is a big refactor, because we also had the same problem with the RPC mode (we were ignoring stream errors and not restarting).

I'm using a blocking queue similar to how we do in the Sync resolver now to handle each even in a simple while loop, and restart the stream on errors.